### ISSUE TITLE
Error with self-intersect Polygon. 

### ISSUE THREAD

**Issue by Rub21** [osmlab/osmlint#29]
I am trying to build [crossing buildings validator](https://github.com/osmlab/osmlint/issues/24), but I got error on turf with self-intersect Polygon:
https://gist.github.com/anonymous/a540a1f9a1d94210d2d8
http://www.openstreetmap.org/way/63020994#map=19/43.72569/7.39558

The implemented function is: 
`turf.intersect(poly1, poly2);`

I found this ticket, https://github.com/Turfjs/turf/issues/92 which say to use [geojsonhint](https://github.com/mapbox/geojsonhint) but it does not work, even I've  tested with https://github.com/craveprogramminginc/GeoJSON-Validation, but does not catch the error as well 

maybe there are  other way to fix it or avoid on analyze this polygon. 
```
/home/ruben/apps/osmlint/node_modules/turf/node_modules/turf-intersect/node_modules/jsts/lib/jsts.js:1436
rlayOp.overlayOp(this.geom[0],this.geom[1],opCode);}catch(ex){throw savedExcep
                                                                    ^
Error
    at Object.<anonymous> (/home/ruben/apps/osmlint/node_modules/turf/node_modules/turf-intersect/node_modules/jsts/lib/jsts.js:21:414)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/ruben/apps/osmlint/node_modules/turf/node_modules/turf-intersect/node_modules/jsts/index.js:2:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)

```
cc. @morganherlocker @mourner @aaronlidman @geohacker

**Comment by morganherlocker** [comment]
@Rub21 [turf.kinks](http://turfjs.org/static/docs/module-turf_kinks.html) can detect self intersections.

**Comment by Rub21** [comment]
thank @morganherlocker , it work fine !
